woudc_data_registry/processing.py

```diff
@@ -178,44 +178,151 @@ def process_data(self, infile, verify_only=False):
+        LOGGER.debug('Validating station data')
+        station = {
+            'identifier': self.data_record.platform_id,
+            'name': self.data_record.platform_name,
+            'country_id': self.data_record.platform_country
+        }
+
+        LOGGER.debug('Validating station id...')
+        results = self.registry.query_multiple_fields(
+            Station, station, ['identifier'])
+        if results:
+            LOGGER.debug("Validated with id:'{}'".format(
+                self.data_record.platform_id))
+        else:
+            msg = 'Station {} not found in registry'.format(
+                self.data_record.platform_id)
             LOGGER.error(msg)
             raise ProcessingError(msg)
 
-        LOGGER.debug('Validating contributor')
-        contributors = self.registry.query_distinct(
-            Contributor.identifier)
+        LOGGER.debug('Validating station name...')
+        fields = ['identifier', 'name']
+        results = self.registry.query_multiple_fields(Station, station, fields)
+        if results:
+            LOGGER.debug("Validated with name:'{}' for id:'{}'".format(
+                self.data_record.platform_name, self.data_record.platform_id))
+        else:
+            msg = 'Station name:"{}" did not match data for id:"{}"'.format(
+                self.data_record.platform_id, self.data_record.platform_id)
+            LOGGER.error(msg)
+            raise ProcessingError(msg)
 
-        if self.data_record.data_generation_agency not in contributors:
-            msg = 'Contributor {} not found in registry'.format(
-                self.data_record.data_generation_agency)
+        LOGGER.debug('Validating station country...')
+        fields = ['identifier', 'country_id']
+        results = self.registry.query_multiple_fields(Station, station, fields)
+        if results:
+            LOGGER.debug("Validated with country:'{}' for id:'{}'".format(
+                self.data_record.platform_country,
+                self.data_record.platform_id))
+        else:
+            msg = "Station country:'{}' did not match data for id:'{}'".format(
+                self.data_record.platform_country,
+                self.data_record.platform_id)
             LOGGER.error(msg)
             raise ProcessingError(msg)
 
-        # TODO: validate station
-        # TODO: validate instrument
+        LOGGER.debug('Validating instrument')
+        self.instruments = self.registry.query_distinct(Instrument.identifier)
+        instrument_added = False
+        instrument = [self.data_record.instrument_name,
+                      self.data_record.instrument_model,
+                      self.data_record.instrument_number,
+                      self.data_record.platform_id,
+                      self.data_record.content_category]
+        instrument_id = ':'.join(instrument)
+        if instrument_id not in self.instruments:
+            instrument[2] = str(int(instrument[2]))
```

Comment:
Why the double cast here?

Reply:
This is to check for the old format by casting it into a integer then creating a instrument_id with it as a string